upgrade_p3_to_p4.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+katello-service stop
+
+# Upgrade puppet 3 to puppet 4
+yum -y localinstall https://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm # puppet 4!!!
+yum remove -y puppet-server # remove old style puppet master
+yum -y install puppetserver #this upgrades puppet, replaces agent too
+
+# Copy puppet environments
+cp -rfp /etc/puppet/environments/* /etc/puppetlabs/code/environments
+
+# Restore puppet 3 info - ssl certs and cache data
+mv /var/lib/puppet/ssl /etc/puppetlabs/puppet
+mv /var/lib/puppet/foreman_cache_data /opt/puppetlabs/puppet/cache/
+
+# Disable HTTP puppet to free up port
```

Comment:
It's noted in http://projects.theforeman.org/projects/foreman/wiki/Upgrading_from_Puppet_3_to_4, but it seems like something a puppet run should clear up. I'll try it without this line to see if the installer fixes it for us.